src/baseline.rs

```diff
@@ -152,12 +180,21 @@ fn generate_rustdoc(
     target_root: PathBuf,
     crate_source: CrateSource,
 ) -> anyhow::Result<PathBuf> {
-    let name = crate_source.name();
-    let version = crate_source.version();
-
-    let (build_dir, cache_dir, cached_rustdoc) = match crate_source {
+    let name = crate_source.name()?;
+    let version = crate_source.version()?;
+
+    let crate_identifier = format!(
+        "{}-{}-{}",
+        match crate_source {
+            CrateSource::Registry { .. } => "registry",
+            CrateSource::ManifestPath { .. } => "local",
```

Comment:
```suggestion
            // Identifiers of manifest-based crates cannot be used for caching,
            // since they probably correspond to a specific (and unknown) gitrev and git state
            // so cached entries cannot be checked to see if they are a match or not.
            CrateSource::ManifestPath { .. } => "local",
```
Just a good reminder for our future selves, once this code is no longer "warm" in our minds.